src/popover/types.js

```diff
@@ -161,6 +161,24 @@ export type PopoverPrivateStateT = {
   isMounted: boolean,
 };
 
+export type ArrowStylePropsArgT = {
+  $arrowOffset: OffsetT,
+  $placement: TetherPlacementT,
+};
+
+export type BodyStylePropsArgT = {
+  $isAnimating: boolean,
+  $isOpen: boolean,
+  $popoverOffset: OffsetT,
+  $placement: TetherPlacementT,
+  $showArrow: boolean,
+};
+
+export type InnerStylePropsArgT = {};
+/*
+* Can't use Intersection types because of https://github.com/facebook/flow/issues/7946
```

Comment:
spreading with exact types may work for you https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVBLAtgBzgJwBdkwBDAZzACUBTUgY2KnzizAHJ87H2BudGgA88RMIQCeOGmCgZ85YgF4wAbwA+qMGADWNcQEYAXGAX4MAOwDmAGlRqAvv1RCRxCVJM16ccwBMwyuqaOnoATMamFjZ2jgLCBG6S0gBGcIQAFgGqwQB0ebLyhLZaeTnkXj6+trGgYM4AbjT44hlRYBiUcNpgCBgZcACuxBaETeWMGD45zvGi7ilpmYGoAJC6BoaRVrZrYZuEZtuosShxrmJJYN64PjTmStTchDkAwqx45neEACpJADypDJgABkqkWTQA0npjKRzOJ7AA+JxQAbmCY+MCWGiEN43T73AAUAEpjNcPl9slouIQBvhzGAcCwcJRFAiwH9fBh6giVIy4MycuD8FD4bUmix8GB0k1pAhpfTAZkOu17mMvIRJvT5ig-sBOdz+PZUN5zAowLicFksTj3rdCUSnC4EmBfDQoKQBjBiMSAgiCX8LSEDIoAET6ENB0Kh0IRoUi0Ph4AIh1AA

Reply:
we will need to maintain `SharedStylePropsArgT` before merging this change

Reply:
@chasestarr What you mean 'maintain SharedStylePropsArgT'. `SharedStylePropsArgT` is present here https://github.com/qqingvarqq/baseui/blob/1c9ff18a174067a7edc35968b7c971290c7b57c4/src/popover/types.js#L182
if need i can replace 
```
export type SharedStylePropsArgT = {
  $arrowOffset: OffsetT,
  $isAnimating: boolean,
  $isOpen: boolean,
  $popoverOffset: OffsetT,
  $placement: TetherPlacementT,
  $showArrow: boolean,
};
```
with 
```
export type SharedStylePropsArgT = {
...$Exact<ArrowStylePropsArgT>,
...$Exact<BodyStylePropsArgT>,
};
```

Reply:
ah sorry - i saw it commented and thought it was removed

Reply:
the second option you have may be ideal

Reply:
@chasestarr updated